VideoLocker/src/main/java/org/edx/mobile/view/UserProfileFragment.java

```diff
@@ -0,0 +1,252 @@
+            getAccountTask.execute();
+        }
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        if (null != getAccountTask) {
+            getAccountTask.cancel(true);
+        }
+    }
+
+    @Override
+    public void onDestroyView() {
+        super.onDestroyView();
+        viewHolder = null;
+    }
+
+
+    private void setAccount(@Nullable final Account account) {
+        this.account = account;
+        if (null == viewHolder) {
+            return;
+        }
+        if (null == account) {
+            viewHolder.profileHeaderContent.setVisibility(View.GONE);
+            viewHolder.profileBodyContent.setVisibility(View.GONE);
+            viewHolder.loadingIndicator.setVisibility(View.VISIBLE);
+
+        } else {
+            viewHolder.profileHeaderContent.setVisibility(View.VISIBLE);
+            viewHolder.profileBodyContent.setVisibility(View.VISIBLE);
+            viewHolder.loadingIndicator.setVisibility(View.GONE);
+
+            final RequestManager requestManager = Glide.with(viewHolder.profileImage.getContext());
+            requestManager
+                    .load(account.getProfileImage().getImageUrlFull())
+                    .thumbnail(requestManager.load(account.getProfileImage().getImageUrlSmall()))
+                    .into(viewHolder.profileImage);
+
+            if (account.requiresParentalConsent() || account.getAccountPrivacy() == Account.Privacy.PRIVATE) {
+                viewHolder.limitedView.setVisibility(View.VISIBLE);
+                viewHolder.limitedView.setText(isViewingOwnProfile
+                        ? R.string.profile_sharing_limited_by_you
+                        : R.string.profile_sharing_limited_by_other_user);
+                viewHolder.languageContainer.setVisibility(View.GONE);
+                viewHolder.locationContainer.setVisibility(View.GONE);
+            } else {
+                viewHolder.limitedView.setVisibility(View.GONE);
+                if (account.getLanguageProficiencies().isEmpty()) {
+                    viewHolder.languageContainer.setVisibility(View.GONE);
+                } else {
+                    viewHolder.languageContainer.setVisibility(View.VISIBLE);
+                    viewHolder.languageText.setText(
+                            new Locale.Builder()
+                                    .setLanguage(account.getLanguageProficiencies().get(0).getCode())
+                                    .build()
+                                    .getDisplayName());
+                }
+
+                if (TextUtils.isEmpty(account.getCountry())) {
+                    viewHolder.locationContainer.setVisibility(View.GONE);
+                } else {
+                    viewHolder.locationContainer.setVisibility(View.VISIBLE);
+                    viewHolder.locationText.setText(
+                            new Locale.Builder()
```

Comment:
`Locale.Builder` was introduced in API Level 21. An imminent FC at this line on pre lollipop device.